### PR TITLE
Use prettier rtfd theme

### PR DESCRIPTION
**Pull request by Baschdl**
Separate requirements.txt is also used by readthedocs

### CODE REVIEW

**Review by bjrne — CHANGES_REQUESTED** [review]
Throws errors:
```bash
$ make html
Running Sphinx v3.1.2

Exception occurred:
  File "/usr/lib/python3.8/importlib/__init__.py", line 118, in import_module
    if name.startswith('.'):
AttributeError: module 'sphinx_rtd_theme' has no attribute 'startswith'
The full traceback has been saved in /tmp/sphinx-err-bou_3p5h.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:20: html] Error 2
```

**Review by bjrne — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -27,6 +27,8 @@ COPY backend/requirements.txt /backend/requirements.txt
 RUN pip3 install -r requirements.txt
 COPY backend/requirements.prod.txt /backend/requirements.prod.txt
 RUN pip3 install -r requirements.prod.txt
+COPY docs/requirements.readthedocs.txt /backend/requirements.readthedocs.txt
```
Comment (bjrne) [inline Dockerfile:30]: Oh no, you really needed to go to sleep :see_no_evil:

**Review by bjrne — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (Baschdl) [reply Dockerfile:30]: Oh no, you really need to learn Docker 🙈

**Comment by Baschdl** [comment]
@maltezacharias Concerning https://github.com/match4everyone/match4everything/pull/131/commits/a26eb9f08bc334613f32b827cde92387fff896e5 and the current implementation of the `.dockerignore`, I have the feeling we introduced quite some complexity by design which we can document to make it easier but I find it questionable if the person which sees an error because of this reads our hint in the right moment. Why don't we ignore everything we don't want but by default include everything? I think that would make it easier for external users which would probably only add a toplevel folder once in a lifetime 😉

**Review by bjrne — APPROVED** [review]

**Comment by maltezacharias** [comment]
Best practice is to keep the docker build context as small as possible and if we go the "only exclude some directories way" that would mean that you will always have to remember to exclude every new file you add that won't be needed for the docker build and I'm pretty sure that's not gonna happen. 

Using the "exclude all, but ..." approach means that your Docker build will fail unless you adressed the file -> Keeping the build context down to only the required files. The .dockerignore right now is already a mix to make errors less likely: "Include all of frontend and backend except for node_modules, pycache and ..." You should only need to adapt the .dockerignore when you change the architecture of the docker containers, e.g. introducing a new intermediate build container or something like that. We could add a hint to the Dockerfile to check the .dockerignore. Then again that is pretty basic Docker knowledge. I feel that is not needed but I won't be against it. I will also gladly write a section in the docs detailing the Docker build design and the design choices that went in there.

That is also no guarantee as @bjrne seems to have included the whole docs dir without using it at all. Please revert that change or explain why it was necessary.

**Comment by maltezacharias** [comment]
EDIT @Baschdl added docs, I read the comments wrong. Still why?

**Pull request merged**